enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,392 @@
+
+Yet, CUS can use any other DIY or just simple Prometheus/Prometheus Operator or vendor (e.g Grafana Cloud or anything else too) to do similar.
+
+### Risks and Mitigations
+
+Let's describe the worst case. What if the Observability Team or/and AppSRE fail to Provide a Managed Observability Service with Required Scalability and reliability (in time)?
+
+Let’s be honest, it’s not an easy job. We are aware of challenges here and the hardest things were presented many times:
+
+![h](assets/hypershift-chall.png)
+
+To the best of our knowledge, we think we can do it. We were preparing the teams to solve those challenges for the last 3 years or so (since the CoreOS acquisition). This is because:
+
+* Service Delivery organization provides amazing support service for Kubernetes, infrastructure, required SaaS services and CI/CD.
+* The monitoring team is active in the development of almost all projects and key dependencies used in the product (Observatorium, Prometheus, Thanos, Cortex, Alertmanager, Loki, etc).
+* The majority part of the monitoring team was designed to be on-call for our workload services too in order to provide stricter SLO and better operationality of our software.
+* Thanos and Loki were designed for high scale and reliability.
+* We have already successfully run the Telemeter service with hundreds of millions of series for last 3 years.
+* We optimized new hires for the Observability Platform team for skillsets to solve those problems. We are hiring for the 2-3 slots.
+
+Mitigations:
+
+* We are growing the dev team even more.
+* We are doubling down the development on performance and scalability. We already have designs to improve the biggest bottlenecks of Thanos.
+* In the extremely worst case: This design assumes API-driven workflows. This allows easy change of Observability Product if anything goes wrong. If our Observability Product effort totally collapses, OSD can within days (and millions $$) switch to any other vendors that support Loki APIs and Remote Write API and Prometheus/Loki read APIs. For example:
+    * Our Grafana Cloud partner
+    * Amazon Managed Prometheus and Grafana
+    * Google Managed Prometheus
+    * Logz.io
+    * … many more
+
+### Migration
+
+N/A. HyperShift is a new project.
+
+### Documentation Required
+
+User facing docs:
+
+* HyperShift client monitoring and how to configure metrics remote write (push API) to desired backend.
+
+Internal docs:
+
+* RHOBS entry-points and usage.
+
+### Open Questions
+
+### FAQ
+
+Some common questions gathered during initial proposal discussions.
+
+#### What About Layered Service / Addons Monitoring?
+
+This was discussed and solved by [Monitoring Stacks](monitoring-stack-operator.md). For HyperShift context, Addons requirements are no different to what we assume Customer would want to do / use. So in the context of this enhancement Addons are just another CUS.
```

Comment:
I can see why this would be a non-goal above. It's easier to say they are just "Customer workloads". 
However, I don't think this is the right attitude if we want addons to succeed as a whole.
I'm thinking internal (RH) here, where there's this RHOBS thing that is solving a lot of problems.
Maybe this isn't the right forum, but why wouldn't addons use that service?

Reply:
Thanks, I see your point, I don't think that is the message that is here. I never said Addons are never part of RHOBS. You are already part of it and using it, so there is definitely such direction. 

We only try to descope too many topics from one proposal. We like it or not our requirements are really not different than customer ones for addons observability (by what we gathered so far). Addons are first citizens as SD org in RHOBS, but AFAIK you can deploy the same way to HyperShift + this enhancement AND to normal OSD and that is the point here, no? (: 

Sorry if you got it as "addons" are less important. They are not. Is there a way I could improve the language here?

Reply:
That makes sense.
Maybe all that's needed here is saying that RHOBS would still be the intended metrics & alerting solution for RH managed services on Hypershift as per https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/monitoring-stack-operator.md) ?